### PR TITLE
Restructure deploy logic for env for consistency

### PR DESCRIPTION
**Pull request by SoManyHs**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/deploy/cloudformation/stack/env_test.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+		},
+		"should return a well formed environment": {
+			mockStack: mockEnvironmentStack("arn:aws:cloudformation:eu-west-3:902697171733:stack/project-env", "project/env"),
+			expectedEnv: archer.Environment{
+				Name:        mockDeployInput.Name,
+				Project:     mockDeployInput.Project,
+				Prod:        mockDeployInput.Prod,
+				AccountID:   "902697171733",
+				Region:      "eu-west-3",
+				RegistryURL: "902697171733.dkr.ecr.eu-west-3.amazonaws.com/project/env",
```
Comment (kohidave) [inline internal/pkg/deploy/cloudformation/stack/env_test.go]: Kill it!

File: internal/pkg/deploy/cloudformation/env.go [hunk 2]
```diff
@@ -1,127 +1,50 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package cloudformation provides functionality to deploy archer resources with AWS CloudFormation.
```
Comment (kohidave) [inline internal/pkg/deploy/cloudformation/env.go]: Can we get rid of this? Don't need no mo'

File: internal/pkg/deploy/cloudformation/stack/env.go [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+
+	// EnvTemplatePath is the path where the cloudformation for the environment is written
+	EnvTemplatePath = "environment/cf.yml"
+
+	// EnvOutputECRKey is the CF Output Key Name for the ECR Repo Name
+	EnvOutputECRKey = "ECRRepositoryName"
+
+	envParamToolsAccountPrincipal = "ToolsAccountPrincipalARN"
+
+	ecrURLFormatString = "%s.dkr.ecr.%s.amazonaws.com/%s"
```
Comment (kohidave) [inline internal/pkg/deploy/cloudformation/stack/env.go]: Can kill this? We don't need anymore.

File: internal/pkg/deploy/cloudformation/stack/env.go [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+	EnvParamProjectNameKey = "ProjectName"
+
+	// EnvParamEnvNameKey is the CF Param Key Name for providing the environment name
+	EnvParamEnvNameKey = "EnvironmentName"
+
+	// EnvTemplatePath is the path where the cloudformation for the environment is written
+	EnvTemplatePath = "environment/cf.yml"
+
+	// EnvOutputECRKey is the CF Output Key Name for the ECR Repo Name
+	EnvOutputECRKey = "ECRRepositoryName"
```
Comment (kohidave) [inline internal/pkg/deploy/cloudformation/stack/env.go]: Can kill this? We don't need anymore.

File: internal/pkg/deploy/cloudformation/stack/env.go [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+
+	createdEnv := archer.Environment{
+		Name:      e.Name,
+		Project:   e.Project,
+		Prod:      e.Prod,
+		Region:    stackARN.Region,
+		AccountID: stackARN.AccountID,
+	}
+
+	if stackOutputs[EnvOutputECRKey] != "" {
```
Comment (kohidave) [inline internal/pkg/deploy/cloudformation/stack/env.go]: Can kill this? We don't need anymore.

File: internal/pkg/deploy/cloudformation/stack/env_test.go [hunk 6]
```diff
@@ -0,0 +1,166 @@
...
+		})
+	}
+}
+func mockEnvironmentStack(stackArn, ecrOutput string) *cloudformation.Stack {
+	return &cloudformation.Stack{
+		StackId: aws.String(stackArn),
+		Outputs: []*cloudformation.Output{
+			&cloudformation.Output{
+				OutputKey:   aws.String(EnvOutputECRKey),
+				OutputValue: aws.String(ecrOutput),
```
Comment (kohidave) [inline internal/pkg/deploy/cloudformation/stack/env_test.go]: We don't need this guy anymore either!

File: internal/pkg/deploy/cloudformation/env.go (hunk 2 above)
Comment (SoManyHs) [reply internal/pkg/deploy/cloudformation/env.go]: need which part?

Comment (SoManyHs) [reply internal/pkg/deploy/cloudformation/env.go]: NEVERMIND

**Pull request merged**